Add information dialog for rules #1494

lottieo opened the pull request:
## Description

Added an information dialog for rules.

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

Closes #1494

lottieo committed:
Add rules dialog #1494

lottieo committed:
Merge branch 'develop' into add-information-dialog-for-rules-#1494

lottieo committed:
Add rule examples with accordions #1494

lottieo committed:
Correct deletion rule example #1494

lottieo committed:
Add unit test #1494

lottieo committed:
Complete unit test and update snapshot

lottieo committed:
Restyle information dialog content #1494

lottieo committed:
Add e2e test #1494

lottieo committed:
Merge branch 'develop' into add-information-dialog-for-rules-#1494

lottieo committed:
Update rules snapshots #1494

joshdimanteto reviewed (changes requested):
Work well, just a couple minor comments  
- The last sentence should be at the top below the opening sentence so user sees the explanation first before reading the examples. 
     
- The examples are a bit unclear, can you make it more distinguishable of what is a system type and usage status. (maybe you can give show a template example to explain the format e.g. `source system type -> destination system type -> usage status`  )

lottieo committed:
Reformat and update rules dialog #1494

joshdimanteto reviewed (commented):
LGTM, you just need to fix the unit and e2e tests

lottieo committed:
Update snapshot and e2e test #1494

joelvdavies commented on src/settings/rules/rulesInformationDialog.component.tsx:
```diff
@@ -0,0 +1,125 @@
+            the items usage status to 'In Use'."
+        />
+        <AccordionContent
+          summary="Example: Storage → Storage → Source Usage Status"
```
I feel this one might be confusing? It wont appear in the table as its always the case regardless of the rules. I have never really considered it as a rule itself though there was a time we thought about it. Perhaps it can just be the text? e.g. `Items can always be moved between systems of the same type, which would not change the usage status.`

lottieo committed:
Update snapshot and e2e test #1494

Removed example moving rule for moving items between two of the same system type as this is not a defined rule

lottieo committed:
Fix e2e test #1494

joshdimanteto reviewed (approved):
LGTM

joelvdavies commented:
@lottieo Just a note, if there are two people on the review, we usually wait for both to approve before merging. No worries though, I can see you addressed my comment, thanks!

lottieo commented:
> @lottieo Just a note, if there are two people on the review, we usually wait for both to approve before merging. No worries though, I can see you addressed my comment, thanks!

My mistake, thanks for letting me know!
